### PR TITLE
Refactor algorithms with simple api

### PR DESCRIPTION
**Pull request by joseph-torsney**
Refactored the algorithms mentioned in commits to use `mantid.simpleapi` instead of `self.createChildAlgorithm` 

**To test:**
Check they still work as expected (or not if the unit tests passing are satisfactory)

associated with #21008 

---

#### Reviewer ####

Please comment on the following ([full description](http://developer.mantidproject.org/ReviewingAPullRequest.html)):

##### Code Review #####

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there is GUI work does it follow the [GUI standards](http://developer.mantidproject.org/Standards/GUIStandards.html)?
- If there are changes in the release notes then do they describe the changes appropriately?

##### Functional Tests #####

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.

### CODE REVIEW

File: Framework/PythonInterface/plugins/algorithms/EnggFitPeaks.py [hunk 1]
```diff
@@ -8,6 +8,7 @@
 
 from mantid.kernel import *
 from mantid.api import *
+from mantid.simpleapi import *
```
Comment (darominski) [inline Framework/PythonInterface/plugins/algorithms/EnggFitPeaks.py:11]: In L13 in the new solution the simpleapi is already imported as mantid. Then, either the new import replaces the previous solution (I believe that'd be preferred ) and all references to simpleapi through mantid.algorithm_name should be refactored, or the new call to Fit should be done through mantid.fit.

File: Framework/PythonInterface/plugins/algorithms/EnggFitPeaks.py [hunk 2]
```diff
@@ -264,24 +265,22 @@ def _fit_single_peak(self, expected_center, initial_params, wks, wks_index):
...
+        fit_output = Fit(
+            Function=fit_function,
+            InputWorkspace=wks,
+            WorkspaceIndex=wks_index,
+            CreateOutput=True,
+            StartX=startx,
+            EndX=endx
+        )
+
+        return fit_output.OutputParameters, fit_output.OutputChi2overDoF
```
Comment (darominski) [inline Framework/PythonInterface/plugins/algorithms/EnggFitPeaks.py:283]: This does work nicely, however leaves `Fit` output workspaces in the ADS that were not there in the earlier solution. If this additional output was not intended, then perhaps the `Fit` algorithm should also have `StoreInADS=False` .

File: Framework/PythonInterface/plugins/algorithms/EnggFitTOFFromPeaks.py [hunk 3]
```diff
@@ -92,25 +93,23 @@ def _fit_dSpacingTOF(self, fitted_peaks_table):
...
-        convert_tbl_alg.setProperty('InputWorkspace', fitted_peaks_table)
-        convert_tbl_alg.setProperty('ColumnX', 'dSpacing')
-        convert_tbl_alg.setProperty('ColumnY', 'X0')
-        convert_tbl_alg.execute()
-        d_tof_conversion_ws = convert_tbl_alg.getProperty('OutputWorkspace').value
+        d_tof_conversion_ws = ConvertTableToMatrixWorkspace(
+            InputWorkspace=fitted_peaks_table,
+            ColumnX='dSpacing',
+            ColumnY='X0'
+        )
```
Comment (darominski) [inline Framework/PythonInterface/plugins/algorithms/EnggFitTOFFromPeaks.py:100]: This also leaves workspaces in the ADS, which is probably not intended. Adding` StoreInADS=False` should restore previous behaviour.

File: Framework/PythonInterface/plugins/algorithms/EnggFitTOFFromPeaks.py [hunk 4]
```diff
@@ -92,25 +93,23 @@ def _fit_dSpacingTOF(self, fitted_peaks_table):
...
-
-        tzero = param_table.cell('Value', 0)  # A0
-        difc = param_table.cell('Value', 1)  # A1
-        difa = param_table.cell('Value', 2)  # A2
+        fit_output = Fit(
+            Function='name=Quadratic',
+            InputWorkspace=d_tof_conversion_ws,
+            WorkspaceIndex=0,
+            CreateOutput=True,
+        )
```
Comment (darominski) [inline Framework/PythonInterface/plugins/algorithms/EnggFitTOFFromPeaks.py:108]: And this call also leaves workspaces in the ADS, which is probably(?) not intended. Adding` StoreInADS=False` should restore previous behaviour.

**Review by darominski — CHANGES_REQUESTED** [review]
It is nice to see the refactoring to more Python native use of algorithms. There are just a few things to add to avoid excessive number of workspaces ending up in the ADS, which was not the case in the previous solution. 
Other calls to algorithms in `EnggFitPeaks` could be refactored in the same way, namely: FindPeaks (L298), CreateWorkspace (L391), ConvertUnits (L401), CreateEmptyTableWorkspace(L460). In addition, the use of `mantid.` to refer to algorithms there as well as the other import of mantid.simpleapi as sapi throughout the code of `EnggFitPeaks`  could be harmonized at the same time.

**Comment by joseph-torsney** [comment]
You're right, I didn't intend the new calls to store things in the ADS, thanks for catching that!

**Comment by DavidFair** [comment]
As this is a maintenance piece which has risk for little gain this close to release, I'm going to move the milestone to 6.2 if that's okay?

**Review by darominski — APPROVED** [review]
Apologies for such a delay in the approval. I somehow missed a notification for the commits with resolved build issues. Looks good now!

**Review by DavidFair — APPROVED** [review]
Looking at the PR dates it took a while, though nobodies fault but this is finally getting merge
I think it's a good improvement to readability and it helps a lot with clearer error messages whilst developing :+1:

**Pull request merged**